src/java/boa/compiler/visitors/CodeGeneratingVisitor.java

```diff
@@ -1668,6 +1817,24 @@ public void visit(final FunctionType n) {
 
 	/** {@inheritDoc} */
 	@Override
+	public void visit(final FixPType n) {
+		if (!(n.type instanceof BoaFunction))
```

Comment:
this check isnt needed in codegen - if it passed typecheck it should be fine here